### PR TITLE
Fix segfault in decompress_chunk for chunks with dropped columns

### PR DESCRIPTION
**Pull request by svenklemm**
This patch fixes a segfault in decompress_chunk for chunks with dropped
columns. Since dropped columns don't exists in the compressed chunk
the values for those columns were undefined in the decompressed tuple
leading to a segfault when trying to build the heap tuple.

### CODE REVIEW

**Review by k-rus — COMMENTED** [review]
Can a test be added for this case?

File: tsl/src/compression/compression.c [hunk 1]
```diff
@@ -1061,6 +1060,14 @@ decompress_chunk(Oid in_table, Oid out_table)
...
+		/*
+		 * We need to make sure decompressed_is_nulls is in a defined state. While this
+		 * will get written for normal columns it will not get written for dropped columns
+		 * since dropped columns don't exist in the compressed chunk so we initiallize
+		 * with true here.
+		 */
+		memset(decompressor.decompressed_is_nulls, true, out_desc->natts);
+
 		Datum *compressed_datums = palloc(sizeof(*compressed_datums) * in_desc->natts);
 		bool *compressed_is_nulls = palloc(sizeof(*compressed_is_nulls) * in_desc->natts);
```
Comment (pmwkaa) [inline tsl/src/compression/compression.c:1072]: Do we need the same here?

File: tsl/src/compression/compression.c [hunk 2]
```diff
@@ -1061,6 +1060,14 @@ decompress_chunk(Oid in_table, Oid out_table)
 			.decompressed_datums = palloc(sizeof(Datum) * out_desc->natts),
```
Comment (pmwkaa) [inline tsl/src/compression/compression.c:1060]: Don't we need to init this field as well in such case? Was thinking about something like `calloc` analogue for memory contexts, if such thing exists

**Review by pmwkaa — APPROVED** [review]
Is it possible to test it?

File: tsl/src/compression/compression.c (hunk 2 above)
Comment (svenklemm) [reply tsl/src/compression/compression.c:1060]: we could do palloc0 here but you are not supposed to look at datum when is_null is true

**Review by k-rus — APPROVED** [review]
It would be great to have a test.

File: tsl/src/compression/compression.c (hunk 1 above)
Comment (svenklemm) [reply tsl/src/compression/compression.c:1072]: no since those will be set by heap_deform_tuple

**Review by gayyappan — APPROVED** [review]

**Pull request merged**